mobile/android/android-emulator-services.ts

```diff
@@ -223,45 +223,64 @@ class AndroidEmulatorServices implements Mobile.IAndroidEmulatorServices {
 
 		client.on('data', (data: any) => {
 			output += data.toString();
+
+			let name = this.getEmulatorNameFromClientOutput(output);
+			// old output should look like:
+			// Android Console: type 'help' for a list of commands
+			// OK
+			// <Name of image>
+			// OK
+
+			// new output should look like:
+			// Android Console: type 'help' for a list of commands
+			// OK
+			// a\u001b[K\u001b[Dav\u001b[K\u001b[D\u001b[Davd\u001b...
+			// <Name of image>
+			// OK
+
+			if (name && !future.isResolved()) {
+				future.return(name);
+			}
+
 			client.end();
 		});
-		client.on('end', () => {
-			//output should look like:
-			//Android Console: type 'help' for a list of commands
-			//OK
-			//<Name of image>
-			//OK
-			//
-			let name: string;
-			let foundOK = false;
-			let lines: string[] = output.split(EOL);
-			// find line between OK
-			_(lines).each((line: string) => {
-				if(foundOK) {
-					name = line.trim();
-					return false;
-				} else if(line.match(/^OK/)) {
-					foundOK = true;
-				}
-			}).value();
 
-			future.return(name);
-		});
 		return future;
 	}
 
+	private getEmulatorNameFromClientOutput(output: string): string {
+		let lines: string[] = output.split(EOL);
+		let name: string;
+
+		let firstIndexOfOk = _.indexOf(lines, "OK");
+
+		if (firstIndexOfOk < 0) {
+			return null;
+		}
+
+		let secondIndexOfOk = _.indexOf(lines, "OK", firstIndexOfOk + 1);
+
+		if (secondIndexOfOk < 0) {
```

Comment:
you do not need this check. In case `firstIndexOfOk` is at least 0, adding 1 to it will never result in negative value